src/main/solution_space/path_finder/path_finder_v_3.py

```diff
@@ -73,7 +73,7 @@ def __find_same_tree_in_graph(self, user_anon_tree: AnonTree, user_canon_tree: a
             if graph_anon_tree:
                 next_anon_trees = [AnonTree.get_item_by_id(id) for id in graph_anon_tree.next_anon_trees_ids]
                 self.write_candidates_info_to_file(next_anon_trees,
-                                                   f'{IPathFinder.get_file_prefix_by_user_tree(user_anon_tree, self.candidates_file_id)}_same_tree_candidates')
+                                                   f'{self.get_file_prefix_by_user_tree(user_anon_tree, self.candidates_file_id)}_same_tree_candidates')
```

Comment:
You can create a field with `self.get_file_prefix_by_user_tree(user_anon_tree, self.candidates_file_id)` and use it inside methods instead of explicitly passing it because its value doesn't change like you do with `candidates_file_id` (which can be removed in this way)